code/modules/projectiles/gun.dm

```diff
@@ -200,13 +202,15 @@
 		else
 			dual_wielding = FALSE
 
-		if(off_hand && off_hand.can_hit(user))
+		if(!can_dual)
+			dual_wielding = FALSE
+		else if(off_hand && off_hand.can_hit(user))
 			spawn(1)
 			off_hand.Fire(A,user,params)
-	else
-		dual_wielding = FALSE
+		else
+			dual_wielding = FALSE
```

Comment:
This too. It could lead to `dual_wielding` not being reset.

Reply:
fixed this too! 

Thanks for the help, still new, still learning, appreciate all the advice I get